lib/internal/fs/streams.js

```diff
@@ -178,7 +183,12 @@ ReadStream.prototype._read = function(n) {
     return this.push(null);
 
   // the actual read.
+  this[kIsPerformingIO] = true;
   fs.read(this.fd, pool, pool.used, toRead, this.pos, (er, bytesRead) => {
+    this[kIsPerformingIO] = false;
+    // Tell ._destroy() that it's safe to close the fd now.
+    if (this.destroyed) return this.emit(kIoDone, er);
```

Comment:
This is observable when `emit()` is monkey-patched, which isn't entirely uncommon. Not a reason per se not to introduce this pattern (it's pretty elegant) but I thought I'd point it out anyway.